src/plugins/tbin/tbinplugin.cpp

```diff
@@ -0,0 +1,358 @@
+                    for ( int ix = 0; ix < tlayer.layerSize.x; ++ix )
+                    {
+                        Tiled::Cell cell = layer->cellAt( ix, iy );
+                        tbin::Tile ttile;
+                        ttile.staticData.tileIndex = -1;
+
+                        if ( cell.tile() != NULL )
+                        {
+                            ttile.tilesheet = cell.tile()->tileset()->name().toStdString();
+                            if ( cell.tile()->frames().size() == 0 )
+                            {
+                                ttile.staticData.tileIndex = cell.tile()->id();
+                            }
+                            else
+                            {
+
+                                // TODO: Check all frame durations are the same
+                                for ( Tiled::Frame frame : cell.tile()->frames() )
+                                {
+                                    ttile.animatedData.frameInterval = frame.duration;
+                                    tbin::Tile tframe;
+                                    tframe.tilesheet = ttile.tilesheet;
+                                    tframe.staticData.tileIndex = frame.tileId;
+                                    ttile.animatedData.frames.push_back( tframe );
+                                }
+                            }
+                        }
+                        tlayer.tiles.push_back( ttile );
+                    }
+                }
+                tiledToTbinProperties(layer, tlayer.props );
+                tmap.layers.push_back( std::move( tlayer ) );
+                tileLayerIdMap[ tmap.layers.back().id ] = &tmap.layers.back();
+            }
+            else
+            {
+                throw std::invalid_argument( "Only object and tile layers supported." );
+            }
+        }
+
+        for ( Tiled::ObjectGroup* objs : objGroups )
+        {
+            for ( Tiled::MapObject* obj : objs->objects() )
+            {
+               if ( obj->name() != "TileData" || obj->size().width() != 1 || obj->size().height() != 1 ||
+                    obj->position().x() != std::floor( obj->position().x() ) || obj->position().y() != std::floor( obj->position().y() ) )
+                   continue;
+
+                tbin::Layer* tiles = tileLayerIdMap[ objs->name().toStdString() ];
+                int idx = static_cast< int >( obj->position().x() + obj->position().y() * tiles->layerSize.x );
+                tiledToTbinProperties( obj, tiles->tiles[ idx ].props );
+            }
+        }
+
+        std::ofstream file( fileName.toStdString(), std::ios::trunc | std::ios::binary );
+        if ( !file )
+        {
+            mError = tr( "Could not open file for writing" );
+            return false;
+        }
+        tmap.saveToStream( file );
+        file.close();
+    }
+    catch ( std::exception& e )
+    {
+        mError = tr((std::string("Exception: ") + e.what()).c_str());
```

Comment:
Passing a dynamic string into `tr` is not going to work, since the `lupdate` tool will not be able to extract the source string this way. It will work however, if the possible strings are wrapped elsewhere with [`QT_TR_NOOP`](http://doc.qt.io/qt-5/qtglobal.html#QT_TR_NOOP). Something like the following may work:
```c++
throw std::invalid_argument(QT_TR_NOOP("Only object and tile layers supported."));
```
and then
```c++
mError = tr("Exception: %1").arg(tr(e.what()));
```